test/src/integration-query-condition.cc

```diff
@@ -0,0 +1,1386 @@
+      std::numeric_limits<float>::epsilon());
+
+  // testing (8,3)
+  REQUIRE(a_data_read_2[1] == a_data_read[142]);
+  REQUIRE(
+      fabs(b_data_read_2[1] - b_data_read[142]) <
+      std::numeric_limits<float>::epsilon());
+
+  // testing (9,3)
+  REQUIRE(a_data_read_2[2] == a_data_read[162]);
+  REQUIRE(
+      fabs(b_data_read_2[2] - b_data_read[162]) <
+      std::numeric_limits<float>::epsilon());
+
+  // testing (10,3)
+  REQUIRE(a_data_read_2[3] == a_data_read[182]);
+  REQUIRE(
+      fabs(b_data_read_2[3] - b_data_read[182]) <
+      std::numeric_limits<float>::epsilon());
+
+  query2.finalize();
+  array2.close();
+
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+}
+
+TEST_CASE(
+    "Test read for sparse arrays (dups allowed) with range within tile for "
+    "rows dimension, "
+    "across tile for col dimension, with query condition",
+    "[query][query-condition][apply-sparse]") {
+  std::srand(static_cast<uint32_t>(time(0)));
+  Context ctx;
+  VFS vfs(ctx);
+
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+
+  std::vector<int> a_data_read(400);
+  std::vector<float> b_data_read(400);
+
+  setup(ctx, TILEDB_SPARSE, true, a_data_read, b_data_read);
+
+  Array array2(ctx, array_name, TILEDB_READ);
+  Query query2(ctx, array2);
+
+  QueryCondition qc(ctx);
+  float val = 4.0f;
+  qc.init("b", &val, sizeof(float), TILEDB_LT);
+
+  std::vector<int> a_data_read_2(8);
+  std::vector<float> b_data_read_2(8);
+  int range[] = {2, 3};
+  int range1[] = {7, 10};
+  query2.add_range("rows", range[0], range[1])
+      .add_range("cols", range1[0], range1[1]);
+  query2.set_layout(TILEDB_UNORDERED)
+      .set_data_buffer("a", a_data_read_2)
+      .set_data_buffer("b", b_data_read_2);
+  query2.set_condition(qc);
+  query2.submit();
+  auto table2 = query2.result_buffer_elements();
+  REQUIRE(table2.size() == 2);
+  REQUIRE(table2["a"].first == 0);
+  REQUIRE(table2["a"].second == 4);
+  REQUIRE(table2["b"].first == 0);
+  REQUIRE(table2["b"].second == 4);
+
+  // Ensure that the same data is in each array (global order)
+  std::vector<float> unordered_read_vals;
+  unordered_read_vals.push_back(b_data_read[26]);  //(2,2)
+  unordered_read_vals.push_back(b_data_read[46]);  //(3,2)
+  unordered_read_vals.push_back(b_data_read[28]);  //(2,3)
+  unordered_read_vals.push_back(b_data_read[48]);  //(3,3)
+
+  std::vector<float> b_data_read_2_copy;
+  for (size_t i = 0; i < 4; i++) {
+    b_data_read_2_copy.push_back(b_data_read_2[i]);
+  }
+
+  for (size_t i = 0; i < 4; ++i) {
+    REQUIRE(a_data_read_2[i] == 1);
+    REQUIRE(
+        fabs(b_data_read_2_copy[i] - unordered_read_vals[i]) <
+        std::numeric_limits<float>::epsilon());
+  }
+
+  query2.finalize();
+  array2.close();
+
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+}
+
+TEST_CASE(
+    "Test read for sparse arrays (dups allowed) with ranges across tiles on "
+    "both dimensions, "
```

Comment:
NIT: fix string to be on 2 lines? :)

Reply:
I see this a few places... I'm guessing clang-format is the culprit

Reply:
Yes, this is definitely clang format oof